gcp/ubuntu-2304/docker-gcc-make.pkr.hcl

```diff
@@ -0,0 +1,81 @@
+packer {
+  required_plugins {
+    googlecompute = {
+      version = ">= 1.0.0"
+      source = "github.com/hashicorp/googlecompute"
+    }
+  }
+}
+
+variable "version" {
+  type = string
+}
+
+variable "project" {
+  type = string
+}
+
+variable "zone" {
+  type = string
+}
+
+locals {
+  source_image = "ubuntu-2304-lunar-amd64-v20230613"
+
+  # System dependencies required for Aspect Workflows
+  install_packages = [
+    # fuse will be optional in future releases although highly recommended for better performance
+    "fuse",
+    # (Optional) zip is required if any tests create zips of undeclared test outputs
+    # For more information about undecalred test outputs, see https://bazel.build/reference/test-encyclopedia
+    "zip",
+    # Additional deps on top of minimal
+    "docker.io",
+    "g++",
+    "make",
+  ]
+
+  # We'll need to tell systemctl to start these when the image boots next.
+  enable_services = [
+    "docker.service",
+  ]
+}
+
+source "googlecompute" "image" {
+  project_id = "${var.project}"
+  image_family = "aspect-workflows-ubuntu-2304"
```

Comment:
I think we want the family to change here. Otherwise someone fetching the latest image in the family may get something that isn't `docker+gcc+make`.

Reply:
right. family should include the variant then?

Reply:
Yes I think so. I'm not sure if we've been consistent in that way across our images.

Reply:
It's not strictly necessary since `family` is just a grouping and we can decide what that means. The image is fully qualified on its own.

Reply:
the AWS way to get the latest is to use a regex like `aspect-workflows-ubuntu-2304-docker-gcc-make-*`. Guessing the typical equivalent on GCP is to use the "family" name?

Reply:
Yeah. There isn't a way to match on the name like that in GCP (last I checked).

Reply:
Gotcha. Updated all the families to include the variant name. After this lands I I think I'll bump the version to 1.2.0 and release the whole lot and then update our use sites to use the latest versions.